test/functional/toolchains/cmake/cmakedeps2/test_cmakeconfigdeps_new.py

```diff
@@ -1572,3 +1572,43 @@ def build(self):
     assert "find_package(matrix)" in c.out
     assert "target_link_libraries(... matrix::matrix)" in c.out
     assert "Conan: Target declared imported INTERFACE library 'matrix::matrix'" in c.out
+
+
+@pytest.mark.tool("cmake", "3.27")
+def test_dependendecy_multi_filename():
+    tc = TestClient()
+    find_name = "foo"
+    conanfile = textwrap.dedent(f"""
+    from conan import ConanFile
+
+    class Pkg(ConanFile):
+        name = "pkg"
+        version = "0.1"
+        def package_info(self):
+            self.cpp_info.set_property("cmake_extra_file_names", ["{find_name}", "extra_name2"])
+    """)
+    cml = textwrap.dedent(f"""
+    cmake_minimum_required(VERSION 3.27)
+    project(app)
+    find_package({find_name} CONFIG REQUIRED)
+    if ({find_name}_FOUND)
+        message("Found {find_name}!")
+        if (TARGET {find_name}::{find_name})
+            message("Found target {find_name}::{find_name}!")
+        endif()
+
+        if (TARGET pkg::pkg)
+            message("Found original target pkg::pkg!")
+        endif()
```

Comment:
Does the file name affect somehow the target name? I don't get this either.

Reply:
No, this is to potentially show users in the future that even if the file name is different, only the default target name for pkg is generated.

This comes from a perceived issue that users might expect conan to declare foo::foo because the find name has changed when using the default-generated Conan target name, not realizing that the target takes the name from the _package_ name, and not the file name